### PR TITLE
[FEAT] Implement core process flow for execution

### PR DESCRIPTION
**Pull request by LeaYeh**
1. Handle AND, OR, PIPE and subshell operation in general way
    2. Make sure subshell and pipeline could get individule working env
    3. Wait for exit code with the correct command table
    4. Add final command table for execution
    5. Handle external and builtin command behavior
    6. Propagate general error through the whole process and exit minishell

### CODE REVIEW

File: source/backend/executor/executor_utils.c [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+	if (!cmd_name)
+		return (NULL);
+	if (ft_strchr(cmd_name, '/') && access(cmd_name, F_OK) == 0)
+		return (ft_strdup(cmd_name));
+	all_path = get_all_path(envp);
+	if (!all_path)
+		return (NULL);
+	i = 0;	
+	while (all_path && all_path[i])
```
Comment (itislu) [inline source/backend/executor/executor_utils.c:78]: `all_path` already get's checked for `NULL` before and doesn't need to be checked for every iteration of the while loop.

**Review by itislu — CHANGES_REQUESTED** [review]

**Review by itislu — CHANGES_REQUESTED** [review]

File: source/backend/executor/handle_simple_cmd.c [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+#include "defines.h"
+#include "utils.h"
+#include "clean.h"
+#include "debug.h"
+
+void	exec_external_cmd(t_shell *shell, t_final_cmd_table *final_cmd_table)
+{
+	// TODO: need to bind the fds to stdin and stdout, if bind failed then exit
+	if (!check_executable(final_cmd_table->cmd_name))
+		ft_clean_and_exit_shell(shell, shell->exit_code, NULL);
```
Comment (itislu) [inline source/backend/executor/handle_simple_cmd.c:23]: `final_cmd_table` does not get free'd

**Review by itislu — CHANGES_REQUESTED** [review]

**Comment by LeaYeh** [comment]
Now I truly believe something going wrong

**Comment by itislu** [comment]
Yuup, we should run the tester locally with this branch and monitor the system.

**Review by itislu — APPROVED** [review]

**Pull request merged**